app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,73 @@
+require_relative '../services/cart_details'
+
+class CartsController < ApplicationController
+  after_action :cart_to_session, except: [:show_cart, :apply_coupon]
+
+  COUPON = 'RUBY'
```

Comment:
Just for information - should be implemented through model with setup in admin panel.